sagemaker-python-sdk/keras_script_mode_pipe_mode_horovod/tensorflow_keras_CIFAR10.ipynb

```diff
@@ -0,0 +1,618 @@
+    "\n",
+    "In this tutorial, we will train a deep CNN to recognize these images.\n",
+    "\n",
+    "We'll compare file mode, pipemode datasets and distributed training with Horovod"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set up the environment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import sagemaker\n",
+    "from sagemaker import get_execution_role\n",
+    "\n",
+    "sagemaker_session = sagemaker.Session()\n",
+    "\n",
+    "role = get_execution_role()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Download the CIFAR-10 dataset\n",
+    "Downloading the test and training data will take around 5 minutes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python generate_cifar10_tfrecords.py --data-dir ./data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a training job using the sagemaker.TensorFlow estimator, running locally\n",
+    "To test that the code will work in SageMaker, we'll first use SageMaker local"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!/home/ec2-user/sample-notebooks/sagemaker-python-sdk/tensorflow_distributed_mnist/setup.sh"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.tensorflow import TensorFlow\n",
+    "\n",
+    "import subprocess\n",
+    "instance_type = 'local'\n",
+    "\n",
+    "if subprocess.call('nvidia-smi') == 0:\n",
+    "    ## Set type to GPU if one is present\n",
+    "    instance_type = 'local_gpu'\n",
+    "    \n",
+    "local_hyperparameters = {'epochs': 5, 'batch-size' : 256}\n",
+    "\n",
+    "source_dir = os.path.join(os.getcwd(), 'source_dir')\n",
+    "estimator = TensorFlow(entry_point='cifar10_keras_main.py',\n",
+    "                       source_dir=source_dir,\n",
+    "                       role=role,\n",
+    "                       framework_version='1.12.0',\n",
+    "                       py_version='py3',\n",
+    "                       hyperparameters=local_hyperparameters,\n",
+    "                       train_instance_count=1, train_instance_type=instance_type)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "local_inputs = {'train' : 'file://'+os.getcwd()+'/data/train', 'validation' : 'file://'+os.getcwd()+'/data/validation', 'eval' : 'file://'+os.getcwd()+'/data/eval'}\n",
+    "estimator.fit(local_inputs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Runing on SageMaker cloud"
```

Comment:
General comment for all headings: use simple present tense, as in previous headings. So "Running on SageMaker cloud"-->"Run on SageMaker cloud"